scripts/src/java/org/oppia/android/scripts/common/BazelClient.kt

```diff
@@ -144,13 +144,37 @@ class BazelClient(private val rootDirectory: File, private val commandExecutor:
   fun runCoverageForTestTarget(bazelTestTarget: String): List<String>? {
     val instrumentation = bazelTestTarget.split(":")[0]
     val computeInstrumentation = instrumentation.split("/").let { "//${it[2]}/..." }
-    val coverageCommandOutputLines = executeBazelCommand(
-      "coverage",
-      bazelTestTarget,
-      "--instrumentation_filter=$computeInstrumentation"
+    val targetParts = bazelTestTarget.split(":")
+    val targetPath = "${targetParts[0]}:*"
+    val targetName = targetParts[1]
+
+    val buildRule = executeBazelCommand(
+      "query",
+      "attr(name, $targetName, $targetPath)",
+      "--output=build",
     )
-    return parseCoverageDataFilePath(coverageCommandOutputLines)?.let { path ->
-      File(path).readLines()
+
+    val hasShardCount = buildRule.any { "shard_count" in it }
+    if (hasShardCount) {
+    executeBazelCommand(
+        "test",
+        "--collect_code_coverage",
+        "--combined_report=lcov",
```

Comment:
Can we perhaps just use this attribute for all runs instead of only when there's a sharded test?

Reply:
I was initially a bit hesitant with using this (`bazel test --collect_code_coverage --combined_report=lcov`) method since it was saving it to one common dir and though we handle things sequentially now, I'm still unsure of what all exceptions or edge cases, coverage runs would have for different files.

Just in a scenario with 10 files running in a shard if file1 ran successfully and file2 too ran successfully but with no .dat produced or any unknown exception, now trying to extract the .dat from the common location would result in mis representation of coverage.

The right way is,
This shouldn't happen at all or should be handled if this ever occurs, but I'm just not sure if it would or might have the possibility. So to be just always sure we have **coverage.dat created specific to its own files/targets** printed right in the terminal for its own run would have the advantages or 
1. to figure out instantly if .dat is not produced making it explicit to handle any unknown cases that might occur 
2. to be sure each run corresponds to its own .dat 

ultimately even shard configured coverage runs too need to be fool proof in this scenario, but what kind of edge case could occur is just unknown to me at this point. 

What would you suggest @BenHenning should we stick to just one common command bazel test --collect_code_coverage or as 2 separate commands.

Reply:
Thanks Ben, with this fix - https://github.com/oppia/oppia-android/pull/5480/files#diff-1e798d9f6f6e6059ff0d747cc961c02e0ba9bf4bf588dfe9db710be512d0c63dR158, now it just has one single command to execute the bazel coverage.

```
executeBazelCommand(
  "coverage",
  bazelTestTarget,
  "--instrumentation_filter=$computeInstrumentation"
)
```

Reply:
I think the latest version is probably the cleanest (one per file/shard and then we combine them when necessary). It gives us a lot more control rather than relying on Bazel behavior (which may not always be entirely understandable when it comes to merging the dat files).